### PR TITLE
[Windows] Avoid crash doing Application.Current.CloseWindow of hidden Window

### LINKED ISSUE #22386: [WinUI]  Application.Current.CloseWindow Crash When AppWindow.Hide is called in .Net Maui 9.0 Prev 3
### Description

basically, I am trying to hide a window first then follow by a close and this is working in .net8 until I move to .net9. The close method only works if I remove the hide method.

[MauiApp2.zip](https://github.com/dotnet/maui/files/15304898/MauiApp2.zip)

### Steps to Reproduce

1. Unzip the attached file and open the solution
2. Click on Open Window on (Window A)

### Link to public reproduction project repository

_No response_

### Version with bug

9.0.0-preview.3.10457

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.3 GA

### Affected platforms

Windows

### Affected platform versions

_No response_

### Did you find any workaround?

nil

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Avoid crash doing Application.Current.CloseWindow of hidden Window.

### Issues Fixed

Fixes #22386

### CODE REVIEW

File: src/Core/src/Handlers/Application/ApplicationHandler.Windows.cs [hunk 1]
```diff
@@ -19,7 +19,20 @@ public partial class ApplicationHandler : ElementHandler<IApplication, UI.Xaml.A
...
+				var platformWindow = window.Handler?.PlatformView as Window;
+
+				if (platformWindow is not null)
+				{
+					var handle = WinRT.Interop.WindowNative.GetWindowHandle(platformWindow);
+					var id = UI.Win32Interop.GetWindowIdFromWindow(handle);
+					var appWindow = UI.Windowing.AppWindow.GetFromWindowId(id);
+
+					// Cannot close an already closed, or disposed Window.
+					if (appWindow is not null && appWindow.IsVisible)
```
Comment (Foda) [inline src/Core/src/Handlers/Application/ApplicationHandler.Windows.cs:31]: I think this is fine. `IsVisible` maps to window style of `WS_VISIBLE`. I was worried that minimized windows might not have that set, but docs say min/restore windows doesn't change that at all :)

**Review by Foda — DISMISSED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22386.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+public class Issue22386 : _IssuesUITest
+{
+	public Issue22386(TestDevice device)
+		: base(device)
+	{ }
+
+	public override string Issue => "[WinUI] Application.Current.CloseWindow Crash When AppWindow.Hide is called in .Net Maui 9.0 Prev 3";
+
+	[Test]
+	public void CloseHiddenWindowNoCrash()
```
Comment (PureWeen) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22386.cs:17]: We need to add a category

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Review by PureWeen — APPROVED** [review]

**Review by Foda — APPROVED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22386.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+{
+	public Issue22386(TestDevice device)
+		: base(device)
+	{ }
+
+	public override string Issue => "[WinUI] Application.Current.CloseWindow Crash When AppWindow.Hide is called in .Net Maui 9.0 Prev 3";
+
+	[Test]
+	[Category(UITestCategories.Window)]
+	public void CloseHiddenWindowNoCrash()
```
Comment (PureWeen) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22386.cs:18]: Test is failing

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Core/src/Handlers/Application/ApplicationHandler.Windows.cs (hunk 1 above)
Comment (mattleibow) [reply src/Core/src/Handlers/Application/ApplicationHandler.Windows.cs:31]: What if the window is hidden? We can't ever close?

You can make windows invisible on Windows for a few reasons - like making a screen clipping app.

**Review by mattleibow — CHANGES_REQUESTED** [review]
Windows need to be hidden sometimes and we need to find a way to correctly close a window if requested.

**Comment by mattleibow** [comment]
This is a bug in the UI tools. Running without the debugger attached works fine. See the issue for more info.

**Pull request closed without merging**